src/adapter/src/coord/sequencer.rs

```diff
@@ -230,6 +230,40 @@ impl<S: Append + 'static> Coordinator<S> {
                 tx.send(self.sequence_show_variable(&session, plan), session);
             }
             Plan::SetVariable(plan) => {
+                // Add notice if database or cluster value does not correspond to a catalog item.
+                if plan.name == "database" || plan.name == "cluster" {
+                    let value = match plan.value.clone() {
+                        mz_sql::ast::SetVariableValue::Ident(ident) => Some(ident.into_string()),
```

Comment:
Per your question in the description, I think this is not valid! Check me on this, but pretty sure `SET database = 'foo'` will give you a `String` instead of an `Ident` here.

Reply:
(Testing both of these cases would be great.)

Reply:
I added test case `Query {"query": "set database = \"materialize\""}` and no notice was produced, then added for  `Query {"query": "set database = \'materialize\'"}` and got the noticed, so that must have produced the literal version.
Do we intentionally support no quote, single quote, and double quote in these commands?

I'll update the code to support both Ident and Literal values.